drv/spartan7-loader-api/src/lib.rs

```diff
@@ -0,0 +1,24 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! API crate for the Spartan7 loader.
+
+#![no_std]
+
+use userlib::sys_send;
+
+/// Token that indicates that the Spartan-7 is running
+///
+/// This token can be passed to peripheral constructors.
+pub struct Spartan7Token(core::marker::PhantomData<()>);
```

Comment:
nitpicky, feel free to ignore me: i don't actually think `PhantomData` is necessary here, this could just be
```suggestion
pub struct Spartan7Token(());
```
as [they're both zero-sized](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=116058fa412d4f13c41b6a3c1cc4750a).

Reply:
Nice, done in 01c2dd1a